topic-operator/src/main/java/io/strimzi/operator/topic/cruisecontrol/CruiseControlHandler.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic.cruisecontrol;
+
+import io.strimzi.api.kafka.model.topic.KafkaTopic;
+import io.strimzi.api.kafka.model.topic.ReplicasChangeStatusBuilder;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.topic.TopicOperatorConfig;
+import io.strimzi.operator.topic.TopicOperatorUtil;
+import io.strimzi.operator.topic.cruisecontrol.CruiseControlClient.TaskState;
+import io.strimzi.operator.topic.metrics.TopicOperatorMetricsHolder;
+import io.strimzi.operator.topic.model.ReconcilableTopic;
+import io.strimzi.operator.topic.model.Results;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static io.strimzi.api.kafka.model.topic.ReplicasChangeState.ONGOING;
+import static io.strimzi.api.kafka.model.topic.ReplicasChangeState.PENDING;
+import static io.strimzi.operator.topic.TopicOperatorUtil.hasReplicasChangeStatus;
+import static io.strimzi.operator.topic.TopicOperatorUtil.topicNames;
+import static java.lang.String.format;
+import static org.apache.logging.log4j.core.util.Throwables.getRootCause;
+
+/**
+ * Handler for Cruise Control requests.
+ */
+public class CruiseControlHandler {
```

Comment:
Personally when there's a state machine like thing going on I find it useful to document that succinctly. Both the states and the allowed transitions. Either in words or in ascii art.